Runtime/Scripts/DebugUtils/BillboardText.cs

```diff
@@ -0,0 +1,47 @@
+using TMPro;
+using UnityEngine;
+
+public class BillboardText : MonoBehaviour
+{
+
+    private Camera _mainCamera;
+    private TextMeshProUGUI _text;
+    private string _content;
+    private float _yOffset;
+    
+    void Awake()
+    {
+        _mainCamera = Camera.main;
+        _text = GetComponentInChildren<TextMeshProUGUI>();
+        if (_content != null)
+        {
+            SetText(_content);
+            SetYOffset(_yOffset);
+        }

```

Comment:
@rcornwall-reg I had it like this because this billboard text is instantiated on the fly and the `SetText` method gets called before the object is awake. So even though it's private and doesn't have a default, it immediately gets set.

But based on your comments, I drastically reduced the complexity of this file now - let me know if that looks good now!